slackbot/main.py

```diff
@@ -18,6 +40,49 @@
 db_port = os.environ.get("DB_PORT")
 db_host = os.environ.get("DB_HOST")
 
+
+
+@app.event("app_mention")
+def handle_mentions(ack, context, body, say, logger, request):
+  mention = body["event"]
+  user= mention["user"]
+  temp_text = mention["text"]
+  channel = mention["channel"]
+  thread_ts = mention["ts"]
+
+  # botのidを削除してテキスト内容のみにする
+  text = re.sub(r'\<.*?\>', '', temp_text)
+
+  # Unix時刻を秒とマイクロ秒に分割
+  seconds = int(float(thread_ts))
+  microseconds = int((float(thread_ts) - seconds) * 1000000)
+
+  # Unixエポック時間を日付に変換
+  date = datetime.datetime.fromtimestamp(seconds) + datetime.timedelta(hours=9)
+
+  # マイクロ秒を追加
+  date_with_microseconds = date + datetime.timedelta(microseconds=microseconds)
+
+  # 日付を文字列に変換
+  date_string = date_with_microseconds.strftime('%Y/%m/%d %H:%M:%S')
+
+  # 送信者の名前をテキストで取得する
+  url = "https://slack.com/api/users.list"
+  response = requests.get(url, headers={"Authorization": "Bearer " + os.environ.get("SLACK_BOT_TOKEN")})
+  users = response.json()["members"]
+  replace_user_dict = {user["id"]:user["profile"]["display_name"] for user in users}
+  sent_user_name = replace_user_dict.get(user)git
+   
+  insert_text=[date_string, sent_user_name, text]
+
+  # google sheets に書き込み
+  last_row_index = len(worksheet.col_values(1)) + 1
+  if not insert_text in worksheet.get_all_values():
```

Comment:
```suggestion
  if insert_text in worksheet.get_all_values() == False:
```
`if not`はあまり利用しない方がいいかなと思います。